docs/sphinx/user_guide/feature/reduction.rst

```diff
@@ -159,3 +159,116 @@ Reduction Policies
 For more information about available RAJA reduction policies and guidance
 on which to use with RAJA execution policies, please see 
 :ref:`reducepolicy-label`.
+
+--------------------------------
+Experimental Reduction Interface
+--------------------------------
+
+An experimental reduction interface is now available that hopes to improve
+upon the current reduction model in RAJA. This interface allows ``RAJA::forall``
+to take optional "plugin-like" objects to extend the execution behaviour 
+of a ``RAJA::forall`` execution context.
+
+The new interface passes ``RAJA::expt::Reduce<OP_TYPE>`` objects as direct
+parameters to ``RAJA::forall`` and provides users with thread-local variables
+of the operating data type to be manipulated indside the lambda. This differs 
+from the current reduction model in which ``RAJA::ReduceOP<REDUCE_POL, T>``
+objects are captured by the RAJA lambda and manipulated directly.
+
+
+RAJA::expt::Reduce
+..................
+::
+
+  double rs = 0.0;
+  double rm = 1e100;
+      
+  RAJA::forall<EXEC_POL> ( Res, Seg, 
+  RAJA::expt::Reduce<RAJA::operators::plus>(&rs),
+  RAJA::expt::Reduce<RAJA::operators::minimum>(&rm),
+  [=] (int i, double& _rs, double& _rm) {
+    _rs += ...
```

Comment:
It would be much clearer and help users if you replaced the ellipses ('...') with actual arguments. For example, here add double* a = ...; before the kernel and use a[i] in the reduction updates in the kernel.

Also, please add a note that 'Res' is a resource object that must be compatible with the exec pol, and 'Seg' is the iteration space object.

Reply:
Done